mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -419,13 +417,27 @@ public LuceneMessageSearchIndex(
             MailboxSessionMapperFactory factory,
             MailboxId.Factory mailboxIdFactory,
             Directory directory,
-            boolean dropIndexOnStart,
+            IndexWriter indexWriter,
             MessageId.Factory messageIdFactory,
             SessionProvider sessionProvider) throws IOException {
         super(factory, ImmutableSet.of(), sessionProvider);
         this.mailboxIdFactory = mailboxIdFactory;
         this.messageIdFactory = messageIdFactory;
         this.directory = directory;
+        this.writer = indexWriter;
+    }
+
+    public LuceneMessageSearchIndex(
```

Comment:
Unhappy with the one - more constructor syndrom

Reply:
I tried to generate one more `IndexWriter` from  `Directory directory` 
but got Lucense exception (lock...bla bla...) 
The creation one more consuctor for input `IndexWriter` -> For quickly bypass it

Reply:
I understand that we wonna share the IndexWriter, +1 for this, but I am unhappy with 1 more constructor.